pallets/corporate-actions/src/distribution.rs

```diff
@@ -0,0 +1,506 @@
+//! Once `expires_at` is reached, however, the remaining amount to distribute is forfeit,
+//! and cannot be claimed by any holder, or pushed to them.
+//! Instead, that amount can be reclaimed by the CAA.
+//!
+//! Before `payment_at` is due, however,
+//! a planned distribution can be cancelled by calling `remove_distribution`.
+//!
+//! ## Overview
+//!
+//! The module provides functions for:
+//!
+//! - Starting a distribution.
+//! - Claiming or pushing benefits of a distribution.
+//! - Reclaiming unclaimed dividends.
+//!
+//! ### Terminology
+//!
+//! - **Currency:** The ticker being distributed to holders as a benefit, e.g., USDC or some such.
+//! - **Payment-at date:** The date at which benefits may be claimed by or pushed to holders.
+//! - **Expires-at date:** The date at which benefits are forfeit, and may be reclaimed by the CAA.
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! - `distribute` starts a capital distribution.
+//! - `claim` claims (pull) a benefit of an active capital distribution on behalf of a holder.
+//! - `push_benefit` pushes a benefit of an active capital distribution to a holder.
+//! - `reclaim` reclaims forfeited benefits of a capital distribution that has expired.
+//! - `remove_distribution` removes a capital distribution which hasn't reached its payment date yet.
+
+use crate as ca;
+use ca::{CAId, Tax, Trait};
+use codec::{Decode, Encode};
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage,
+    dispatch::{DispatchError, DispatchResult},
+    ensure,
+};
+use pallet_asset::{self as asset, checkpoint};
+use pallet_identity as identity;
+use polymesh_common_utilities::{
+    portfolio::PortfolioSubTrait,
+    protocol_fee::{ChargeProtocolFee, ProtocolOp},
+    with_transaction, CommonTrait,
+};
+use polymesh_primitives::{EventDid, IdentityId, Moment, PortfolioId, PortfolioNumber, Ticker};
+use sp_runtime::traits::{CheckedDiv, CheckedMul};
+#[cfg(feature = "std")]
+use sp_runtime::{Deserialize, Serialize};
+use sp_std::prelude::*;
+
+type Asset<T> = asset::Module<T>;
+type Checkpoint<T> = checkpoint::Module<T>;
+type CA<T> = ca::Module<T>;
+type Identity<T> = identity::Module<T>;
+type Portfolio<T> = pallet_portfolio::Module<T>;
+
+/// A capital distribution's various details.
+///
+/// All information contained is used by on-chain logic.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Encode, Decode)]
+pub struct Distribution<Balance> {
+    /// The portfolio to distribute from.
+    pub from: PortfolioId,
+    /// The currency that payouts happen in.
+    pub currency: Ticker,
+    /// Total amount to be distributed.
+    pub amount: Balance,
+    /// Amount left to distribute.
+    pub remaining: Balance,
+    /// Whether the CAA has claimed remaining funds.
+    pub reclaimed: bool,
+    /// An optional timestamp of payout start
```

Comment:
I don't think this is optional?

Reply:
```suggestion
    /// A timestamp of payout start
```
Yep, forgot to update it.